### DISCUSSION TITLE
[Bug] Reserved characters in NAME interact with env.sh during backup?

### DISCUSSION THREAD

**Discussion by Spunkie** [mbround18/valheim-docker#591] [Help!] [answered]
First is an error about `.bash_profile` that logs at the same time regardless of what the server `NAME` is set to. And the 2nd error only pops up if I put a space(` `) and a `!` character in the server `NAME`. For example with `NAME="abc defg!"` the error looks like this:
```shell
==> /home/steam/valheim/logs/auto-backup.out <==
/env.sh: line 7: /home/steam/.bash_profile: No such file or directory
/env.sh: line 20: export: `defg!': not a valid identifier
[Valheim][steam]: Skipping backup, no players are online.
```
If you remove either character `NAME="abcdefg!"` or `NAME="abc defg"` then the error does not happen, but a name with both seems to trigger it. I didn't have time to test for other characters in a server `NAME`, that might cause the same behavior, but I imagine there are some more creeping about.

This is on a stock amazon x86 EC2 instance running Amazon Linux 2, Kernel 5.10.

___
There is also another random error that pops up consistently at boot and shutdown:
```shell
[Valheim][steam]: Running Install...
[Valheim][steam]: Attempting to update before launching the server!
parse error: Invalid numeric literal at line 1, column 6
```
```shell
Backup on shutdown triggered! Running backup tool...
parse error: Invalid numeric literal at line 1, column 6
```

**Comment by mbround18** [comment]
@Spunkie a name with a space and an exclamation mark?

**Reply by Spunkie** [reply]
Ya, as the the valheim server name. The one set with the `NAME` env.

**Reply by mbround18** [reply]
Hmm i could just replace those characters with dashes? tho most names should be unix compliant for the file system which i dont think an exclamation is unix compliant 0.o

**Reply by Spunkie** [reply]
From a user perspective it's pretty unexpected that the valhiem server name interacts with or is used for a token in the file system at all. 

Would that thought `NAME` was submitted by API and used for nothing else, so no worries about what characters to use as long as the steam server browser/valheim renders it.

**Reply by mbround18** [reply]
@Spunkie we could alter it to be the backup gets a name of backup-{timecode} and allow the option of it being overridden?

**Reply by Spunkie** [reply]
Sounds good to me.

**Reply by mbround18** [reply]
@Spunkie i should have a pr up for it tomorrow, first day of my vacation nd got a few errands to run before i can crank it out.

**Reply by Spunkie** [reply]
No rush here, but I really appreciate the update 😁 

Enjoy your vacation 🌴🥥💆

**Reply by DiskCrasher** [reply]
I've noticed that when NAME has a space in it the name that appears in the server listing is truncated. For example, "My Server" gets registered as just "My".

**Reply by mbround18** [reply]
@DiskCrasher  Is your server name quoted? Drop a snippet of your name env variable

**Reply by DiskCrasher** [reply]
@mbround18 Yes, it's quoted:

`NAME: "Shooting Star"`

On startup the log shows:

`[Valheim][steam]: Name: Shooting`

**Reply by mbround18** [reply]
My guess is this is due to the translation between docker and env vars. I can take a look at it and see if i can improve the env var translation.

**Reply by spannerman79** [reply]
@DiskCrasher There is a work around until @mbround18 gets around it.

Double escape the space.

Log:

`[Valheim][steam]: Name: Shooting Star`

en var:

`NAME: "Shooting\\ Star"`

I didn't do anything fancy for testing but for replication purpose;

```docker-compose.yml
version: "3"
services:
  valheim:
    image: mbround18/valheim:latest
    stop_signal: SIGINT
    ports:
      - "2456:2456/udp"
      - "2457:2457/udp"
      - "2458:2458/udp"
    environment:
      PORT: 2456
      NAME: "Shooting\\ Star"
      WORLD: "Dedicated"
      PASSWORD: "Banana Phone"
      TZ: "America/Chicago"
      PUBLIC: 1
    volumes:
      - ./saves:/home/steam/.config/unity3d/IronGate/Valheim
      - ./server:/home/steam/valheim
```

**Reply by mbround18** [reply, marked as answer]
I have quoted the string in the template you no longer need to escape the space. https://github.com/mbround18/valheim-docker/releases/tag/v2.4.0
